packages/mdc-segmented-button/segmented-button/test/component.test.ts

```diff
@@ -0,0 +1,341 @@
+      }
+
+      setAllSelected(segments);
+      adapter.unselectSegment(test_segment_ids.UNSELECTED_SEGMENT_ID);
+      for (let i = 0; i < segments.length; i++) {
+        expect(segments[i].classList.contains(test_css_classes.SELECTED)).toEqual(i !== test_indices.UNSELECTED);
+      }
+
+      component.destroy();
+    });
+
+    it('#unselectSegment unselects no child segment if none is identified', () => {
+      const {root, component, adapter} = setupTest();
+
+      const segments = root.querySelectorAll(test_selectors.SEGMENT);
+      setAllSelected(segments);
+
+      adapter.unselectSegment(test_indices.NOT_PRESENT);
+      for (let i = 0; i < segments.length; i++) {
+        expect(segments[i].classList.contains(test_css_classes.SELECTED)).toBeTrue();
+      }
+
+      adapter.unselectSegment(test_segment_ids.NOT_PRESENT_SEGMENT_ID);
+      for (let i = 0; i < segments.length; i++) {
+        expect(segments[i].classList.contains(test_css_classes.SELECTED)).toBeTrue();
+      }
+
+      component.destroy();
+    });
+
+    it(`#notifySelectedChange emits ${events.CHANGE} with SegmentDetail`, () => {
+      const {component, adapter} = setupTest();
+      const handler = jasmine.createSpy('change handler');
+
+      component.listen(events.CHANGE, handler);
+
+      adapter.notifySelectedChange({
+        index: test_indices.SELECTED,
+        selected: true,
+        segmentId: test_segment_ids.SELECTED_SEGMENT_ID
+      });
+      expect(handler).toHaveBeenCalledWith(jasmine.anything());
+      expect(handler.calls.mostRecent().args[0].detail.index).toEqual(test_indices.SELECTED);
+      expect(handler.calls.mostRecent().args[0].detail.selected).toBeTrue();
+      expect(handler.calls.mostRecent().args[0].detail.segmentId).toEqual(test_segment_ids.SELECTED_SEGMENT_ID);
+
+      component.unlisten(events.CHANGE, handler);
+      component.destroy();
+    });
+  });
+
+  it('#getSelectedSegments returns selected child segments as a SegmentDetail list', () => {
+    const {root, component} = setupTest();
+
+    const segments = root.querySelectorAll(test_selectors.SEGMENT);
+    setAllSelected(segments);
+
+    const selected_segments = component.getSelectedSegments();
+    expect(selected_segments.length).toEqual(segments.length);
+    for (let i = 0; i < selected_segments.length; i++) {
+      expect(selected_segments[i].index).toEqual(i);
+      expect(selected_segments[i].selected).toBeTrue();
+      expect(selected_segments[i].hasOwnProperty('segmentId')).toBeTrue();
+    }
+
+    setAllUnselected(segments);
+    expect(component.getSelectedSegments().length).toEqual(0);
+
+    component.destroy();
+  });
+
+  it('#selectSegment selects identified child segment', () => {
+    const {root, component} = setupTest();
+
+    const segments = root.querySelectorAll(test_selectors.SEGMENT);
+    const selected_segment = segments[test_indices.SELECTED];
```

Comment:
Variables in TypeScript (& JS) should use [camelCase](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined) naming format.